idx/memory/memory.go

```diff
@@ -130,6 +144,69 @@ func (m *MemoryIdx) Update(entry idx.Archive) {
 	m.Unlock()
 }
 
+func (m *MemoryIdx) addTags(def *schema.MetricDefinition) {
+	tags, ok := m.Tags[def.OrgId]
+	if !ok {
+		tags = make(map[string]map[string]map[string]struct{})
+		m.Tags[def.OrgId] = tags
+	}
+	for _, tag := range def.Tags {
+		tagSplits := strings.SplitN(tag, "=", 2)
+		if len(tagSplits) < 2 {
+			// invalid
+			continue
+		}
+
+		tagName := tagSplits[0]
+		tagValue := tagSplits[1]
+
+		if _, ok = tags[tagName]; !ok {
+			tags[tagName] = make(map[string]map[string]struct{})
```

Comment:
maybe we should give these substructures a name, so that it's easier/clearer to create them

Reply:
That might be good, but I can't come up with a good name :) 
The top level structure is currently named `TagIndex`. I suggest the next one could be `TagKey` and it's nested values would be called `TagValue`. So each `TagValue` would then be a set of IDs

Reply:
how about:
```
TagIndex -> TagValues -> MetricDefinitions
```